pipeline/mutate/mutator_hydrator.go

```diff
@@ -104,6 +143,13 @@ func (a *MutatorHydrator) Mutate(r *http.Request, session *authn.AuthenticationS
 		return errors.WithStack(err)
 	}
 
+	if cfg.Cache.Ttl != "" {
```

Comment:
Should we not also respsect `Cache-Control` headers? Maybe the server doesn't want us to cache the requests and sends `Cache-Control: no-cache` or some other cache value?

Reply:
There's actually a http.NetTransport that does that: https://github.com/gregjones/httpcache

Reply:
That library will only work for GET requests. I imagine the hydrator is mostly used where someone is controlling both Oathkeeper and the hydrate service.

Reply:
Ah, I see - yeah I think it makes sense. We can also add this at a later stage anyways as it won't be a breaking change.